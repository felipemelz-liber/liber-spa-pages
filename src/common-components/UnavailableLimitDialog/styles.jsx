import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
