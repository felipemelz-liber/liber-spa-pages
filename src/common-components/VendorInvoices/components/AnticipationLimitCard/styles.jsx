import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
