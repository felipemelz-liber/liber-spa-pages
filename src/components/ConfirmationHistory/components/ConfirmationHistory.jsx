import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import ToastContainer from 'liber-components/components/Toast';
import { fetchConfirmations, setCredentials, updateFilters } from '../actions/index.js';
import ConfirmationHistoryTable from './Table.jsx';
import {
  ConfirmationHistoryContainer,
  LoadingBlock,
  Loading,
  Global,
  ScreenTitle,
  TitleContainer,
} from './ConfirmationHistory.styles.jsx';

const DATE_LOWER = 'q[created_at_date_lteq]';
const DATE_GREATER = 'q[created_at_date_gteq]';
const DATE_FORMAT = 'YYYY-MM-DD';

export const ConfirmationHistory = ({
  token,
  sacadoId,
  onSetCredentials,
  onFetchConfirmations,
  loadingAllPage,
  per,
  onUpdateFilters,
}) => {
  React.useEffect(() => {
    onSetCredentials(token, sacadoId);
    onFetchConfirmations({ token, params: { page: '1', per: '10' } });
  }, []);

  const handleFilter = ({ start, end }) => {
    let newFilters = {};
    if (start && end) {
      newFilters = {
        [DATE_GREATER]: start.format(DATE_FORMAT),
        [DATE_LOWER]: end.format(DATE_FORMAT),
      };
    } else if (start || end) {
      newFilters = {
        [DATE_GREATER]: start ? start.format(DATE_FORMAT) : end.format(DATE_FORMAT),
        [DATE_LOWER]: start ? start.format(DATE_FORMAT) : end.format(DATE_FORMAT),
      };
    }
    onUpdateFilters(newFilters);
    onFetchConfirmations({ token, params: { page: '1', per, ...newFilters } });
  };

  return (
    <>
      <Global />
      {loadingAllPage ? <Loading loop /> : null}
      <LoadingBlock loading={loadingAllPage ? 1 : 0}>
        <ConfirmationHistoryContainer>
          <TitleContainer>
            <ScreenTitle>Histórico de Confirmações</ScreenTitle>
          </TitleContainer>
          <ConfirmationHistoryTable handleFilter={handleFilter} />
        </ConfirmationHistoryContainer>
      </LoadingBlock>
      <ToastContainer />
    </>
  );
};

ConfirmationHistory.propTypes = {
  onSetCredentials: PropTypes.func.isRequired,
  token: PropTypes.string.isRequired,
  sacadoId: PropTypes.number.isRequired,
  onFetchConfirmations: PropTypes.func.isRequired,
  loadingAllPage: PropTypes.bool.isRequired,
  per: PropTypes.number.isRequired,
  onUpdateFilters: PropTypes.func.isRequired,
};

const mapStateToProps = ({ confirmationHistory }) => ({
  loadingAllPage: confirmationHistory.loadingAllPage,
  per: confirmationHistory.pagination.per,
});

const mapDispatchToProps = {
  onFetchConfirmations: fetchConfirmations,
  onSetCredentials: setCredentials,
  onUpdateFilters: updateFilters,
};

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmationHistory);
