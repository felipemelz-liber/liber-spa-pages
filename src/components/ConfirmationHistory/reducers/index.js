import confirmationHistory from './ConfirmationHistoryReducer';

export default {
  confirmationHistory,
};
