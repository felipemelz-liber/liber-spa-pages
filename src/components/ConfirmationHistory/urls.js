const domain = INVESTOR_DASHBORD_API; // eslint-disable-line

export const CONFIRMATIONS_HISTORY_URL = `${domain}/api/v1/sacado/direito_creditorios`;
