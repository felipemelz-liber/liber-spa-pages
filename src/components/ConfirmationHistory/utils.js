import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton, DownloadIcon } from './components/Table.styles.jsx';

export const filtersPropTypes = PropTypes.shape({
  'q[created_at_eq]': PropTypes.string,
  'q[created_at_gteq]': PropTypes.string,
  'q[created_at_lteq]': PropTypes.string,
});

export const CONFIRMATION_HIST_COLUMNS = [
  {
    name: 'slug',
    label: 'Código',
    width: 12,
    align: 'center',
  },
  {
    name: 'value',
    label: 'Valor Total',
    width: 18,
    align: 'right',
    getCellValue: ({ value }) => `R$ ${value}`,
  },
  {
    name: 'created_at',
    label: 'Data de Criação',
    width: 18,
    align: 'right',
  },
  {
    name: 'contract',
    label: 'Contrato',
    width: 25,
    align: 'left',
    // eslint-disable-next-line react/prop-types
    getCellValue: ({ contract = '#' }) => (
      <StyledButton version={2} size="small" href={contract} download>
        <DownloadIcon />
        Download
      </StyledButton>
    ),
  },
];
