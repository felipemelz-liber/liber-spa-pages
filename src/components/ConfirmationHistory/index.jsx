import React from "react";
import PropTypes from "prop-types";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { LiberV4 } from "liber-components";
import hydrateStore from "../store";
import ConfirmationHistory from "./components/ConfirmationHistory.jsx";

const View = ({ token, sacadoId }) => (
  <>
    <Provider store={hydrateStore({})}>
      <ConfirmationHistory token={token} sacadoId={sacadoId} />
    </Provider>
  </>
);

export default View;
