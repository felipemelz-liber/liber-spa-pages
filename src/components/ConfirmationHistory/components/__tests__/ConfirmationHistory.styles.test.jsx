import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as styles from '../ConfirmationHistory.styles';

describe('Testing ConfirmationHistory styles', () => {
  configure({ adapter: new Adapter() });
  const { LoadingBlock, ...uncontrolled } = styles;
  Object.keys(uncontrolled).forEach(style => {
    it(`renders ${style} correctly`, () => {
      const App = uncontrolled[style];
      const wrapper = shallow(<App />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  it('renders LoadingBlock correctly without props', () => {
    const wrapper = shallow(<LoadingBlock />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders LoadingBlock correctly while loading', () => {
    const wrapper = shallow(<LoadingBlock loading={1} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders LoadingBlock correctly while not loading', () => {
    const wrapper = shallow(<LoadingBlock loading={0} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
