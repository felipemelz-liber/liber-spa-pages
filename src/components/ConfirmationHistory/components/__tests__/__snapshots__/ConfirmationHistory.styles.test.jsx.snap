// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfirmationHistory styles renders ConfirmationHistoryContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistory styles renders Global correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Testing ConfirmationHistory styles renders Loading correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
  width: 200px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%);
",
        ],
      },
      "displayName": "Styled(Bunny)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistory styles renders LoadingBlock correctly while loading 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  flex-grow: 1;
  display: flex;
  opacity: ",
          [Function],
          ";
  pointer-events: ",
          [Function],
          ";
  transition: opacity 150ms ease-in-out;
  display: inherit;
  flex-direction: inherit;
  position: inherit;
  min-height: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  loading={1}
/>
`;

exports[`Testing ConfirmationHistory styles renders LoadingBlock correctly while not loading 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  flex-grow: 1;
  display: flex;
  opacity: ",
          [Function],
          ";
  pointer-events: ",
          [Function],
          ";
  transition: opacity 150ms ease-in-out;
  display: inherit;
  flex-direction: inherit;
  position: inherit;
  min-height: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  loading={0}
/>
`;

exports[`Testing ConfirmationHistory styles renders LoadingBlock correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  flex-grow: 1;
  display: flex;
  opacity: ",
          [Function],
          ";
  pointer-events: ",
          [Function],
          ";
  transition: opacity 150ms ease-in-out;
  display: inherit;
  flex-direction: inherit;
  position: inherit;
  min-height: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistory styles renders MdHelpCircleStyled correctly 1`] = `
<styled.div>
  <Styled(HelpCircle) />
</styled.div>
`;

exports[`Testing ConfirmationHistory styles renders ScreenTitle correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 32px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.41;
  letter-spacing: 0.13px;
  text-align: left;
  color: #009dff;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistory styles renders TitleContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  margin-top: 32px;
  margin-bottom: 36px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
